File: Java/Warmup-2/stringMatch.java

public class warmup2 {

	// this function return the number of the positions where they contain the same length 2 substring
	public int stringMatch(String a, String b) {
	
		int count = 0;
		int length = Math.min(a.length(), b.length());
		
		for (int i = 0; i < length - 1; i++) {
			if (a.substring(i, i+2).compareTo(b.substring(i, i+2)) == 0)
				count++;
		}
		
		return count;
		
	}
	
	public static void main(String[] args) {
		
		warmup2 o = new warmup2();

		System.out.println(o.stringMatch("xxcaazz", "xxbaaz"));
		System.out.println(o.stringMatch("abc", "abc"));
		System.out.println(o.stringMatch("abc", "axc"));

	}

}

File: Python/Warmup-2/string_match.py

# this function return the number of the positions where they contain the same length 2 substring 
def string_match(a, b):
	count = 0
	length = min(len(a), len(b))

	for x in range(0,length - 1):
		if a[x:x+2] == b[x:x+2]:
			count += 1

	return count

print(string_match("xxcaazz", "xxbaaz"))
print(string_match("abc", "abc"))
print(string_match("abc", "axc"))
